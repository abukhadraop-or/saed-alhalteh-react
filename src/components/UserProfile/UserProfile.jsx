import React from 'react';
import { useHistory } from 'react-router-dom';
import ProfileImage from './user-profile.style';

function UserProfile() {
  const history = useHistory();
  const userInfo = history.location.state.detail.author;
  return (
    <div>
      <h1>
        username:
        {userInfo.username}
      </h1>
      <ProfileImage src={userInfo.image} alt="personal" />
      <p>
        Bio:
        {userInfo.bio}
      </p>

      <div> HEre we will display posts created by this user</div>
    </div>
  );
}

export default UserProfile;
