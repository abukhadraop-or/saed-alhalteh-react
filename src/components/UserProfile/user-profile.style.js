import styled from 'styled-components';

const ProfileImage = styled.img`
  border-radius: 50%;
  width: 6.25rem;
  height: 6.25rem;
`;

export default ProfileImage;
