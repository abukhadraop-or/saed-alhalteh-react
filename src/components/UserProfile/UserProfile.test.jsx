import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import UserProfile from '.';

describe('<UserProfile /> tests', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<UserProfile />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
