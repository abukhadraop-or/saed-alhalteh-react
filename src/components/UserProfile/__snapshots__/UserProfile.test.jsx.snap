// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> tests matches the snapshot 1`] = `
<h1>
  User Profile 
</h1>
`;
